dockerfiles/grist/ressources/dinum-custom.js

```diff
@@ -0,0 +1,143 @@
+          </p>
+          <button disabled id="fermerPopinMaintenance">Fermer</button>
+        </form>
+    `;
+    document.body.appendChild(dialog);
+    document.getElementById('fermerPopinMaintenance').onclick = function onMessageUnderstood() {
+      localStorage.maintainanceStartDateAgreement = maintainanceStartDate.toISOString();
+    }
+    document.getElementById('jaiCompris').onchange = function (ev) {
+      document.getElementById('fermerPopinMaintenance').disabled = !ev.target.checked;
+    }
+    dialog.showModal();
+  }
+
+  if (Date.now() < maintainanceEndDate.getTime() &&
+    (!localStorage.maintainanceStartDateAgreement || localStorage.maintainanceStartDateAgreement !== maintainanceStartDate.toISOString())) {
+    window.addEventListener('load', showDialog);
+  }
+})();
+
+// ========================
+// END dialog
+// ========================
+
+// ========================
+// START gauffre
+// ========================
+
+window.onload = (event) => {
+  const pathname = window.location.pathname;
+
+  // Gauffre is only inserted on grist home page
+  if (pathname !== '/o/docs/'){
+    return 1;
+  }
+
+
+  waitForElm('.test-top-header').then((elm) => {
+    // Create gauffre button Tag
+    const gristBar = document.getElementsByClassName('test-top-header')[0];
+    const gauffreDiv = document.createElement('div');
+    const gauffreButton = document.createElement('button');
+    const gauffreText = "Les services de La Suite numérique";
+
+    gauffreDiv.className = 'gauffre-container';
+    gauffreButton.type = "button";
+    gauffreButton.className = "lasuite-gaufre-btn lasuite-gaufre-btn--vanilla js-lasuite-gaufre-btn";
+    gauffreButton.title = gauffreText;
+    gauffreButton.text = gauffreText;
+    gauffreDiv.appendChild(gauffreButton);
+    gristBar.insertBefore(gauffreDiv, gristBar.lastChild);
+
+    // Create gauffre script Tag
+    const gauffreScript = document.createElement('script');
+    const head = document.getElementsByTagName('head')[0];
+
+    gauffreScript.id = "lasuite-gaufre-script";
+    gauffreScript.setAttribute('async', true);
+    gauffreScript.setAttribute('defer', true);
+    gauffreScript.src = "https://integration.lasuite.numerique.gouv.fr/api/v1/gaufre.js";
+    head.insertBefore(gauffreScript, head.lastChild);
+  });
+};
+
+// from https://stackoverflow.com/a/61511955
+function waitForElm(selector) {
+  return new Promise(resolve => {
+    if (document.querySelector(selector)) {
+      return resolve(document.querySelector(selector));
+    }
+
+    const observer = new MutationObserver(mutations => {
+      if (document.querySelector(selector)) {
+        observer.disconnect();
+        resolve(document.querySelector(selector));
+      }
+    });
+
+    // If you get "parameter 1 is not of type 'Node'" error, see https://stackoverflow.com/a/77855838/492336
+    observer.observe(document.body, {
```

Comment:
If possible, you may try to reduce the observable scope to a body descendant, to optimize a bit.

Reply:
The dom is constructed from body with _grainXXX classes that I fear we can't assume that this XXX will be generated with the same at every build of the application.
So I think we don't have a better solution for now.